src/common/mod.rs

```diff
@@ -31,6 +31,8 @@ pub const MEBIBYTE: u64 = 1_024 * KIBIBYTE;
 pub const KIBIBYTE: u64 = 1_024 * BYTE;
 pub const BYTE: u64 = 1;
 
+pub const SAMPLE_PERIOD: u64 = 100; 
```

Comment:
Might be better to define this in Cargo.toml

Reply:
If we do frequency based, we can still get accurate counts? I'd be concerned about not hitting the period for rarer events. Specifying a frequency in the config, assuming there's some overhead associated with higher frequencies, seems like it might give more predictable and controllable resource utilization?